src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -564,6 +613,8 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
     let resultingFirstChild : ?Fiber = null;
     let previousNewFiber : ?Fiber = null;
 
+    let knownKeysInDev = null;
```

Comment:
Thank you for explaining. I fixed this in https://github.com/facebook/react/pull/8570/commits/ff66eb6e556ac71fac71a13cddd791ffb8a19cb3.